addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -311,3 +315,58 @@ def _get_iap_search_term(self, email):
         """
         domain = tools.email_domain_extract(email)
         return ("@" + domain) if domain not in iap_tools._MAIL_DOMAIN_BLACKLIST else email
+
+    def _prepare_translations(self, plugin):
+        """
+        Returns translations for terms used in the mail addons, this method can be overridden by other modules to return
+        their related translations.
+
+        for params inside the translated string, only named string params e.g: %(param)s are supported by the plugins,
+        unnamed params like %s are not supported.
+
+        each entry must have the exact same values for it's key and it's translation value in the returned dictionary.
+        e.g: "Company created": _("Company created") is correct and "company_created": ("Company created") is incorrect,
+        this allows for easier and more intuitive manipulation of translated strings from the plugins side.
+
+        :param plugin: name of the plugin asking for translations
+        """
+        translations = {}
+
+        if plugin == "outlook":
+            translations = {
```

Comment:
When reading @std-odoo PR, you have several common items. Couldn't we simplify overall diff and have a
```
if plugin in ('outlook', 'gmail'):
    # blah
if plugin == 'outlook':
    # additional outlook strings
if plugin == 'gmail':
    # additional gmail strings
```

Maybe we could keep a single PR, either this one or Stephane's one, as I think it will be easier to review and test. What do you think ?

Reply:
The problem with the proposed approach I think, is that if we do this, when we modify/add/delete a translated string, we will have to check all plugins, this can be time consuming especially when there are a lots of strings.

Reply:
I would also prefer to keep it separate, so if I change something on gmail, I do not need to check that it do not break a view or is still relevant for outlook...

But if we really want to split the translations, I'll do it in my own PR (keep it like this on yours)